### ISSUE TITLE
PhET-iO instrumentation of Preferences and its dialog

### ISSUE THREAD

**Issue by zepumph** [phetsims/joist#744]
Today during PhET-iO meeting we were discussing how to instrument these preferences.

We feel strongly about two things:
1. We want a researcher to be able to know what preferences were selected during the sim run.
2. We don't want preferences to be set onto another sim from setState.

For now we will achieve this by instrumenting preferences as phetioReadOnly: true, and phetioState:false. This way they show up in the data stream.

**Comment by zepumph** [comment]
What about an instrumented Property that stored whether or not preferences should be stored? Then you can turn it on and off.

**Comment by zepumph** [comment]
- [ ] We should also instrument the Preferences button on the nav bar.

**Comment by arouinfar** [comment]
@zepumph based on our discussion in the last PhET-iO meeting, I think `general.view.soundManager.enhancedSoundEnabledProperty` may need to be deprecated. The "Enhanced Sound" option has been rebranded as "Extra Sounds" in the preferences dialog. You can see it in context in John Travoltage:

![image](https://user-images.githubusercontent.com/8419308/134083716-fac005d7-ae37-43ed-a5ac-690eed66c956.png)

**Comment by zepumph** [comment]
In https://github.com/phetsims/joist/issues/743#issuecomment-921996720 we decided that this issue is blocked by that one. Marking accordingly.

**Comment by zepumph** [comment]
During PhET-iO meeting today, we decided that preferences Properties would not be stateful. If in the future clients desire this feature, then we can discuss making it possible. 

We also decided that it is acceptable and helpful to have data stream capabilities, so that a researcher could look at learners that use voicing or not (for example). This feels helpful to the goals of PhET-iO.



- [ ] In general, we want to customize per feature, and not internal to a feature. So you can't hide a checkbox in the Voicing controls, but you can hide the whole voicing section.

- [ ] Ability to hide tabs in the preferences, can we use the QP below, or do we need new Properties that are instrumented.
  supportsInteractiveHighlights=false
  supportsVoicing=false
  audio=disabled
  supportsSound=false

- [ ] Ensure you can hide the preferences button


- [ ] rename soundManager.enhancedSoundEnabledProperty to soundManager.extraSoundsEnabledProperty
even though the query parameter is public, we can change it. In general, enhancedSound=>extraSounds (new issue)

- [ ] Design meeting after initial implementation.

**Comment by zepumph** [comment]
I got pretty far with initial instrumentation, but ran into a bug when trying to alert a description to aria live. I'll have to come back to this.


<details>

```
Index: phet-io-wrappers/input-record-and-playback/inputRecordAndPlayback.js
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/phet-io-wrappers/input-record-and-playback/inputRecordAndPlayback.js b/phet-io-wrappers/input-record-and-playback/inputRecordAndPlayback.js
--- a/phet-io-wrappers/input-record-and-playback/inputRecordAndPlayback.js	(revision 17e332da4b14b43f1a54f7f00e87ffa278e05ec1)
+++ b/phet-io-wrappers/input-record-and-playback/inputRecordAndPlayback.js	(date 1647380181461)
@@ -75,6 +75,14 @@
         } );
       } );
     }, 10 );
+
+
+    for ( let i = 0; i < sourceFrame.contentWindow.ALL_NODES.length; i++ ) {
+      if ( sourceFrame.contentWindow.ALL_NODES[ i ] !== destinationFrame.contentWindow.ALL_NODES[ i ] ) {
+        debugger;
+      }
+    }
+
   } );
 
   const setActivePropertyUpstreamSim = active => sourceClient.invoke( activePropertyPhetioID, 'setValue', [ active ] );
Index: scenery/js/nodes/Node.ts
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/scenery/js/nodes/Node.ts b/scenery/js/nodes/Node.ts
--- a/scenery/js/nodes/Node.ts	(revision 8084c8fc3d43397a1940e284b69b2b5657428f9b)
+++ b/scenery/js/nodes/Node.ts	(date 1647380181529)
@@ -205,6 +205,7 @@
   'centerY' // {number} - The y-center of this Node's bounds, see setCenterY() for more documentation
 ];
 
+window.ALL_NODES = [];
 // Node options, in the order they are executed in the constructor/mutate()
 const NODE_OPTION_KEYS = [
   'children', // {Array.<Node>}- List of children to add (in order), see setChildren for more documentation
@@ -637,6 +638,10 @@
     super();
 
     this._id = globalIdCounter++;
+    if( this._id === 2006){
+      debugger;
+    }
+    window.ALL_NODES.push( `${this.constructor.name}${this._id}` );
     this._instances = [];
     this._rootedDisplays = [];
     this._drawables = [];
Index: joist/js/A11yButtonsHBox.js
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/joist/js/A11yButtonsHBox.js b/joist/js/A11yButtonsHBox.js
--- a/joist/js/A11yButtonsHBox.js	(revision 6f617114eef353e72bea7319355af3d753269558)
+++ b/joist/js/A11yButtonsHBox.js	(date 1647380543570)
@@ -40,7 +40,7 @@
       const preferencesButton = new NavigationBarPreferencesButton(
         sim.preferencesManager,
         backgroundColorProperty, {
-          tandem: options.tandem.createTandem( 'navigationBarPreferencesButton' )
+          tandem: options.tandem.createTandem( 'preferencesButton' )
         } );
 
       a11yButtons.push( preferencesButton );
Index: joist/js/preferences/PreferencesToggleSwitch.js
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/joist/js/preferences/PreferencesToggleSwitch.js b/joist/js/preferences/PreferencesToggleSwitch.js
--- a/joist/js/preferences/PreferencesToggleSwitch.js	(revision 6f617114eef353e72bea7319355af3d753269558)
+++ b/joist/js/preferences/PreferencesToggleSwitch.js	(date 1647381019873)
@@ -50,8 +50,8 @@
         trackFillRight: '#64bd5a'
       },
 
-      // phet-io - opting out of Tandems for now
-      tandem: Tandem.OPT_OUT
+      // phet-io
+      tandem: Tandem.REQUIRED
     }, options );
     assert && assert( options.labelNode === null || options.labelNode instanceof Node, 'labelNode is null or inserted as child' );
     assert && assert( options.descriptionNode === null || options.descriptionNode instanceof Node, 'labelNode is null or inserted as child' );
@@ -75,8 +75,8 @@
       voicingIgnoreVoicingManagerProperties: true,
       voicingNameResponse: options.a11yLabel,
 
-      // tandem - opting out of Tandems for now
-      tandem: Tandem.OPT_OUT
+      // tandem
+      tandem: options.tandem.createTandem( 'toggleSwitch' )
     } ) );
 
     this.addChild( toggleSwitch );
Index: joist/js/preferences/GeneralPreferencesPanel.js
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/joist/js/preferences/GeneralPreferencesPanel.js b/joist/js/preferences/GeneralPreferencesPanel.js
--- a/joist/js/preferences/GeneralPreferencesPanel.js	(revision 6f617114eef353e72bea7319355af3d753269558)
+++ b/joist/js/preferences/GeneralPreferencesPanel.js	(date 1647382221566)
@@ -8,8 +8,8 @@
  */
 
 import merge from '../../../phet-core/js/merge.js';
-import { VoicingRichText } from '../../../scenery/js/imports.js';
-import { VBox } from '../../../scenery/js/imports.js';
+import { VBox, VoicingRichText } from '../../../scenery/js/imports.js';
+import Tandem from '../../../tandem/js/Tandem.js';
 import joist from '../joist.js';
 import joistStrings from '../joistStrings.js';
 import PreferencesDialog from './PreferencesDialog.js';
@@ -23,17 +23,24 @@
 
   /**
    * @param {Object} generalModel - configuration for the Tab, see PreferencesManager for entries
+   * @param {Object} [options]
    */
-  constructor( generalModel ) {
-    super( {
+  constructor( generalModel, options ) {
+
+    options = merge( {
       align: 'left',
       spacing: 40,
 
       // pdom
       tagName: 'section',
       labelTagName: 'h2',
-      labelContent: 'General'
-    } );
+      labelContent: 'General',
+
+      // phet-io
+      tandem: Tandem.REQUIRED
+    }, options );
+
+    super( options );
 
     const panelChildren = [];
     generalModel.simControls && panelChildren.push( new SimControlsPanelSection( generalModel.simControls ) );
Index: joist/js/toolbar/Toolbar.js
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/joist/js/toolbar/Toolbar.js b/joist/js/toolbar/Toolbar.js
--- a/joist/js/toolbar/Toolbar.js	(revision 6f617114eef353e72bea7319355af3d753269558)
+++ b/joist/js/toolbar/Toolbar.js	(date 1647381873479)
@@ -18,11 +18,8 @@
 import stepTimer from '../../../axon/js/stepTimer.js';
 import Matrix3 from '../../../dot/js/Matrix3.js';
 import { Shape } from '../../../kite/js/imports.js';
-import { voicingManager } from '../../../scenery/js/imports.js';
-import { voicingUtteranceQueue } from '../../../scenery/js/imports.js';
-import { Node } from '../../../scenery/js/imports.js';
-import { Path } from '../../../scenery/js/imports.js';
-import { Rectangle } from '../../../scenery/js/imports.js';
+import merge from '../../../phet-core/js/merge.js';
+import { Node, Path, Rectangle, voicingManager, voicingUtteranceQueue } from '../../../scenery/js/imports.js';
 import ButtonNode from '../../../sun/js/buttons/ButtonNode.js';
 import RoundPushButton from '../../../sun/js/buttons/RoundPushButton.js';
 import Tandem from '../../../tandem/js/Tandem.js';
@@ -48,14 +45,20 @@
 
   /**
    * @param {Sim} sim
+   * @param {Object} [options]
    */
-  constructor( sim ) {
+  constructor( sim, options ) {
 
-    super( {
+    options = merge( {
 
       // pdom
-      tagName: 'div'
-    } );
+      tagName: 'div',
+
+      // phet-io
+      tandem: Tandem.REQUIRED
+    }, options );
+
+    super( options );
 
     // @private {BooleanProperty} - Whether or not the Toolbar is enabled (visible to the user)
     this.isEnabledProperty = sim.preferencesManager.toolbarEnabledProperty;
@@ -89,7 +92,9 @@
     // @private {VoicingToolbarItem} - Contents for the Toolbar, currently only controls related to the voicing
     // feature.
     const voicingAlertManager = new VoicingToolbarAlertManager( sim.screenProperty );
-    this.menuContent = new VoicingToolbarItem( voicingAlertManager, sim.lookAndFeel );
+    this.menuContent = new VoicingToolbarItem( voicingAlertManager, sim.lookAndFeel, {
+      tandem: options.tandem.createTandem( 'menuContent' )
+    } );
 
     // icon for the openButton
     const chevronIcon = new DoubleChevron();
Index: joist/js/preferences/PreferencesTabs.js
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/joist/js/preferences/PreferencesTabs.js b/joist/js/preferences/PreferencesTabs.js
--- a/joist/js/preferences/PreferencesTabs.js	(revision 6f617114eef353e72bea7319355af3d753269558)
+++ b/joist/js/preferences/PreferencesTabs.js	(date 1647382546235)
@@ -8,15 +8,9 @@
  */
 
 import Property from '../../../axon/js/Property.js';
+import merge from '../../../phet-core/js/merge.js';
 import StringUtils from '../../../phetcommon/js/util/StringUtils.js';
-import { FocusHighlightPath } from '../../../scenery/js/imports.js';
-import { KeyboardUtils } from '../../../scenery/js/imports.js';
-import { Voicing } from '../../../scenery/js/imports.js';
-import { PressListener } from '../../../scenery/js/imports.js';
-import { Line } from '../../../scenery/js/imports.js';
-import { Node } from '../../../scenery/js/imports.js';
-import { Rectangle } from '../../../scenery/js/imports.js';
-import { Text } from '../../../scenery/js/imports.js';
+import { FocusHighlightPath, KeyboardUtils, Line, Node, PressListener, Rectangle, Text, Voicing } from '../../../scenery/js/imports.js';
 import Tandem from '../../../tandem/js/Tandem.js';
 import joist from '../joist.js';
 import joistStrings from '../joistStrings.js';
@@ -34,14 +28,21 @@
   /**
    * @param {PreferencesTab[]} supportedTabs - list of tabs the Dialog should include
    * @param {EnumerationDeprecatedProperty.<PreferencesDialog.PreferenceTab>} selectedPanelProperty
+   * @param {Object} [options]
    */
-  constructor( supportedTabs, selectedPanelProperty ) {
-    super( {
+  constructor( supportedTabs, selectedPanelProperty, options ) {
+    options = merge( {
 
+      // pdom
       tagName: 'ul',
       ariaRole: 'tablist',
-      groupFocusHighlight: true
-    } );
+      groupFocusHighlight: true,
+
+      // phet-io
+      tandem: Tandem.REQUIRED
+    }, options );
+
+    super( options );
 
     // @private {null|Node} - A reference to the selected and focusable tab content so that we can determine which
     // tab is next in order when cycling through with alternative input.
@@ -52,13 +53,14 @@
 
     // @private {Tab[]}
     this.content = [];
-    const addTabIfSupported = ( preferenceTab, titleString ) => {
-      _.includes( supportedTabs, preferenceTab ) && this.content.push( new Tab( titleString, selectedPanelProperty, preferenceTab ) );
+    const addTabIfSupported = ( preferenceTab, titleString, tandemName ) => {
+      const tandem = options.tandem.createTandem( tandemName );
+      _.includes( supportedTabs, preferenceTab ) && this.content.push( new Tab( titleString, selectedPanelProperty, preferenceTab, tandem ) );
     };
-    addTabIfSupported( PreferencesDialog.PreferencesTab.GENERAL, generalTitleString );
-    addTabIfSupported( PreferencesDialog.PreferencesTab.VISUAL, visualTitleString );
-    addTabIfSupported( PreferencesDialog.PreferencesTab.AUDIO, audioTitleString );
-    addTabIfSupported( PreferencesDialog.PreferencesTab.INPUT, inputTitleString );
+    addTabIfSupported( PreferencesDialog.PreferencesTab.GENERAL, generalTitleString, 'generalTab' );
+    addTabIfSupported( PreferencesDialog.PreferencesTab.VISUAL, visualTitleString, 'visualTab' );
+    addTabIfSupported( PreferencesDialog.PreferencesTab.AUDIO, audioTitleString, 'audioTab' );
+    addTabIfSupported( PreferencesDialog.PreferencesTab.INPUT, inputTitleString, 'audioTab' );
 
     for ( let i = 0; i < this.content.length; i++ ) {
       this.addChild( this.content[ i ] );
@@ -154,7 +156,7 @@
    * @param {EnumerationDeprecatedProperty.<PreferencesDialog.<PreferencesTab>} property
    * @param {PreferencesDialog.PreferencesTab} value - PreferencesTab shown when this tab is selected
    */
-  constructor( label, property, value ) {
+  constructor( label, property, value, tandem ) {
 
     const textNode = new Text( label, PreferencesDialog.TAB_OPTIONS );
 
@@ -178,7 +180,9 @@
       innerContent: label,
       ariaRole: 'tab',
       focusable: true,
-      containerTagName: 'li'
+      containerTagName: 'li',
+
+      tandem: tandem
     } );
 
     // @public {PreferenceTab}
@@ -188,7 +192,7 @@
       title: label
     } );
 
-    const buttonListener = new PressListener( {
+    const pressListener = new PressListener( {
       press: () => {
         property.set( value );
 
@@ -196,12 +200,12 @@
         this.voicingSpeakNameResponse();
       },
 
-      // phet-io - opting out for now to get CT working
-      tandem: Tandem.OPT_OUT
+      // phet-io
+      tandem: tandem.createTandem( 'pressListener' )
     } );
-    this.addInputListener( buttonListener );
+    this.addInputListener( pressListener );
 
-    Property.multilink( [ property, buttonListener.isOverProperty ], ( selectedTab, isOver ) => {
+    Property.multilink( [ property, pressListener.isOverProperty ], ( selectedTab, isOver ) => {
       textNode.opacity = selectedTab === value ? 1 :
                          isOver ? 0.8 :
                          0.6;
Index: joist/js/toolbar/VoicingToolbarItem.js
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/joist/js/toolbar/VoicingToolbarItem.js b/joist/js/toolbar/VoicingToolbarItem.js
--- a/joist/js/toolbar/VoicingToolbarItem.js	(revision 6f617114eef353e72bea7319355af3d753269558)
+++ b/joist/js/toolbar/VoicingToolbarItem.js	(date 1647381946682)
@@ -8,15 +8,10 @@
  */
 
 import BooleanProperty from '../../../axon/js/BooleanProperty.js';
+import merge from '../../../phet-core/js/merge.js';
 import PlayStopButton from '../../../scenery-phet/js/buttons/PlayStopButton.js';
 import PhetFont from '../../../scenery-phet/js/PhetFont.js';
-import { VoicingText } from '../../../scenery/js/imports.js';
-import { ReadingBlockHighlight } from '../../../scenery/js/imports.js';
-import { voicingManager } from '../../../scenery/js/imports.js';
-import { AlignGroup } from '../../../scenery/js/imports.js';
-import { HBox } from '../../../scenery/js/imports.js';
-import { Node } from '../../../scenery/js/imports.js';
-import { Text } from '../../../scenery/js/imports.js';
+import { AlignGroup, HBox, Node, ReadingBlockHighlight, Text, voicingManager, VoicingText } from '../../../scenery/js/imports.js';
 import Tandem from '../../../tandem/js/Tandem.js';
 import Utterance from '../../../utterance-queue/js/Utterance.js';
 import joist from '../joist.js';
@@ -46,16 +41,21 @@
   /**
    * @param {VoicingToolbarAlertManager} alertManager - generates the alert content when buttons are pressed
    * @param {LookAndFeel} lookAndFeel
+   * @param {Object} [options]
    */
-  constructor( alertManager, lookAndFeel ) {
-
-    super( {
+  constructor( alertManager, lookAndFeel, options ) {
+    options = merge( {
 
       // pdom
       tagName: 'section',
       labelTagName: 'h2',
-      labelContent: toolbarString
-    } );
+      labelContent: toolbarString,
+
+      // phet-io
+      tandem: Tandem.REQUIRED
+    }, options );
+
+    super( options );
 
     const titleTextOptions = {
       font: new PhetFont( 14 ),
@@ -76,7 +76,8 @@
       a11yLabel: titleString,
       toggleSwitchOptions: {
         voicingUtteranceQueue: joistVoicingUtteranceQueue
-      }
+      },
+      tandem: options.tandem.createTandem( 'muteSpeechSwitch' )
     } );
 
     // layout
Index: joist/js/preferences/SoundPanelSection.js
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/joist/js/preferences/SoundPanelSection.js b/joist/js/preferences/SoundPanelSection.js
--- a/joist/js/preferences/SoundPanelSection.js	(revision 6f617114eef353e72bea7319355af3d753269558)
+++ b/joist/js/preferences/SoundPanelSection.js	(date 1647381462367)
@@ -8,11 +8,7 @@
 
 import merge from '../../../phet-core/js/merge.js';
 import StringUtils from '../../../phetcommon/js/util/StringUtils.js';
-import { VoicingRichText } from '../../../scenery/js/imports.js';
-import { VoicingText } from '../../../scenery/js/imports.js';
-import { voicingUtteranceQueue } from '../../../scenery/js/imports.js';
-import { Text } from '../../../scenery/js/imports.js';
-import { VBox } from '../../../scenery/js/imports.js';
+import { Text, VBox, VoicingRichText, VoicingText, voicingUtteranceQueue } from '../../../scenery/js/imports.js';
 import Checkbox from '../../../sun/js/Checkbox.js';
 import soundManager from '../../../tambo/js/soundManager.js';
 import Tandem from '../../../tandem/js/Tandem.js';
@@ -47,12 +43,15 @@
       // PreferencesPanelSection. It is possible that the toggle for Sound can be redundant when Sound
       // is the only Audio feature supported. In that case, control of Sound should go through the
       // "All Audio" toggle.
-      includeTitleToggleSwitch: true
+      includeTitleToggleSwitch: true,
+
+      // phet-io
+      tandem: Tandem.REQUIRED
     }, options );
 
     const soundLabel = new Text( soundsLabelString, PreferencesDialog.PANEL_SECTION_LABEL_OPTIONS );
 
-    const titleNode = new PreferencesToggleSwitch( soundManager.enabledProperty, false, true, {
+    const soundEnabledSwitch = new PreferencesToggleSwitch( soundManager.enabledProperty, false, true, {
       labelNode: soundLabel,
       descriptionNode: new VoicingText( soundDescriptionString, merge( {}, PreferencesDialog.PANEL_SECTION_CONTENT_OPTIONS, {
         readingBlockContent: StringUtils.fillIn( labelledDescriptionPatternString, {
@@ -63,7 +62,8 @@
       toggleSwitchOptions: {
         visible: options.includeTitleToggleSwitch
       },
-      a11yLabel: soundsLabelString
+      a11yLabel: soundsLabelString,
+      tandem: options.tandem.createTandem( 'soundEnabledSwitch' )
     } );
 
     let enhancedSoundContent = null;
@@ -79,7 +79,7 @@
         voicingNameResponse: extraSoundsLabelString,
 
         // phet-io
-        tandem: Tandem.OPT_OUT
+        tandem: options.tandem.createTandem( 'enhancedSoundCheckbox' )
       } );
 
       const enhancedSoundDescription = new VoicingRichText( extraSoundsDescriptionString, merge( {}, PreferencesDialog.PANEL_SECTION_CONTENT_OPTIONS, {
@@ -102,7 +102,7 @@
     }
 
     super( {
-      titleNode: titleNode,
+      titleNode: soundEnabledSwitch,
       contentNode: enhancedSoundContent
     } );
 
Index: joist/js/preferences/InputPreferencesPanel.js
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/joist/js/preferences/InputPreferencesPanel.js b/joist/js/preferences/InputPreferencesPanel.js
--- a/joist/js/preferences/InputPreferencesPanel.js	(revision 6f617114eef353e72bea7319355af3d753269558)
+++ b/joist/js/preferences/InputPreferencesPanel.js	(date 1647382257387)
@@ -8,10 +8,8 @@
 
 import merge from '../../../phet-core/js/merge.js';
 import StringUtils from '../../../phetcommon/js/util/StringUtils.js';
-import { VoicingRichText } from '../../../scenery/js/imports.js';
-import { voicingUtteranceQueue } from '../../../scenery/js/imports.js';
-import { Node } from '../../../scenery/js/imports.js';
-import { Text } from '../../../scenery/js/imports.js';
+import { Node, Text, VoicingRichText, voicingUtteranceQueue } from '../../../scenery/js/imports.js';
+import Tandem from '../../../tandem/js/Tandem.js';
 import joist from '../joist.js';
 import joistStrings from '../joistStrings.js';
 import PreferencesDialog from './PreferencesDialog.js';
@@ -32,17 +30,22 @@
 
   /**
    * @param {Object} inputModel - see PreferencesManager
+   * @param {Object} [options]
    */
-  constructor( inputModel ) {
-    super( {
+  constructor( inputModel, options ) {
+    options = merge( {
 
       // pdom
       tagName: 'div',
       labelTagName: 'h2',
-      labelContent: inputTitleString
-    } );
+      labelContent: inputTitleString,
+
+      tandem: Tandem.REQUIRED
+    }, options );
 
-    const toggleSwitch = new PreferencesToggleSwitch( inputModel.gestureControlsEnabledProperty, false, true, {
+    super( options );
+
+    const gestureControlsEnabledSwitch = new PreferencesToggleSwitch( inputModel.gestureControlsEnabledProperty, false, true, {
       labelNode: new Text( gestureControlsString, PreferencesDialog.PANEL_SECTION_LABEL_OPTIONS ),
       descriptionNode: new VoicingRichText( gestureControlsDescriptionString, merge( {}, PreferencesDialog.PANEL_SECTION_CONTENT_OPTIONS, {
         lineWrap: 350,
@@ -52,11 +55,12 @@
           description: gestureControlsDescriptionString
         } )
       } ) ),
-      a11yLabel: gestureControlsString
+      a11yLabel: gestureControlsString,
+      tandem: options.tandem.createTandem( 'gestureControlsEnabledSwitch' )
     } );
 
     const panelSection = new PreferencesPanelSection( {
-      titleNode: toggleSwitch
+      titleNode: gestureControlsEnabledSwitch
     } );
     this.addChild( panelSection );
 
Index: joist/js/preferences/AudioPreferencesPanel.js
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/joist/js/preferences/AudioPreferencesPanel.js b/joist/js/preferences/AudioPreferencesPanel.js
--- a/joist/js/preferences/AudioPreferencesPanel.js	(revision 6f617114eef353e72bea7319355af3d753269558)
+++ b/joist/js/preferences/AudioPreferencesPanel.js	(date 1647381462364)
@@ -6,9 +6,9 @@
  * @author Jesse Greenberg (PhET Interactive Simulations)
  */
 
-import { HBox } from '../../../scenery/js/imports.js';
-import { Text } from '../../../scenery/js/imports.js';
-import { VBox } from '../../../scenery/js/imports.js';
+import merge from '../../../phet-core/js/merge.js';
+import { HBox, Text, VBox } from '../../../scenery/js/imports.js';
+import Tandem from '../../../tandem/js/Tandem.js';
 import joist from '../joist.js';
 import joistStrings from '../joistStrings.js';
 import PreferencesDialog from './PreferencesDialog.js';
@@ -24,13 +24,20 @@
   /**
    * @param {Object} audioModel - configuration for audio settings, see PreferencesManager
    * @param {BooleanProperty} enableToolbarProperty - whether the Toolbar is enabled
+   * @param {Object} [options]
    */
-  constructor( audioModel, enableToolbarProperty ) {
+  constructor( audioModel, enableToolbarProperty, options ) {
+
+    options = merge( {
+      tandem: Tandem.REQUIRED
+    }, options );
 
     const panelChildren = [];
 
     if ( audioModel.supportsVoicing ) {
-      panelChildren.push( new VoicingPanelSection( audioModel, enableToolbarProperty ) );
+      panelChildren.push( new VoicingPanelSection( audioModel, enableToolbarProperty, {
+        tandem: options.tandem.createTandem( 'voicingPanelSection' )
+      } ) );
     }
 
     if ( audioModel.supportsSound ) {
@@ -41,7 +48,8 @@
       const hideSoundToggle = audioModel.supportsVoicing !== audioModel.supportsSound;
 
       panelChildren.push( new SoundPanelSection( audioModel, {
-        includeTitleToggleSwitch: !hideSoundToggle
+        includeTitleToggleSwitch: !hideSoundToggle,
+        tandem: options.tandem.createTandem( 'soundPanelSection' )
       } ) );
     }
 
@@ -52,7 +60,8 @@
 
     const allAudioSwitch = new PreferencesToggleSwitch( audioModel.simSoundEnabledProperty, false, true, {
       labelNode: new Text( audioFeaturesString, PreferencesDialog.PANEL_SECTION_LABEL_OPTIONS ),
-      a11yLabel: audioFeaturesString
+      a11yLabel: audioFeaturesString,
+      tandem: options.tandem.createTandem( 'allAudioSwitch' )
     } );
 
     const soundEnabledListener = enabled => {
Index: joist/js/preferences/VoicingPanelSection.js
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/joist/js/preferences/VoicingPanelSection.js b/joist/js/preferences/VoicingPanelSection.js
--- a/joist/js/preferences/VoicingPanelSection.js	(revision 6f617114eef353e72bea7319355af3d753269558)
+++ b/joist/js/preferences/VoicingPanelSection.js	(date 1647383310313)
@@ -89,12 +89,17 @@
   /**
    * @param {Object} audioModel - configuration for audio settings, see PreferencesManager
    * @param {BooleanProperty} toolbarEnabledProperty - whether or not the Toolbar is enabled for use
+   * @param {Object} [options]
    */
-  constructor( audioModel, toolbarEnabledProperty ) {
+  constructor( audioModel, toolbarEnabledProperty, options ) {
+
+    options = merge( {
+      tandem: Tandem.REQUIRED
+    }, options );
 
     // the checkbox is the title for the section and totally enables/disables the feature
     const voicingLabel = new Text( voicingLabelString, PreferencesDialog.PANEL_SECTION_LABEL_OPTIONS );
-    const voicingSwitch = new PreferencesToggleSwitch( audioModel.voicingEnabledProperty, false, true, {
+    const voicingEnabledSwitch = new PreferencesToggleSwitch( audioModel.voicingEnabledProperty, false, true, {
       labelNode: voicingLabel,
       descriptionNode: new VoicingText( voicingDescriptionString, merge( {}, PreferencesDialog.PANEL_SECTION_CONTENT_OPTIONS, {
         readingBlockContent: StringUtils.fillIn( labelledDescriptionPatternString, {
@@ -102,14 +107,16 @@
           description: voicingDescriptionString
         } )
       } ) ),
-      a11yLabel: voicingLabelString
+      a11yLabel: voicingLabelString,
+      tandem: options.tandem.createTandem( 'voicingEnabledSwitch' )
     } );
 
     // checkbox for the toolbar
     const quickAccessLabel = new Text( toolbarLabelString, PreferencesDialog.PANEL_SECTION_LABEL_OPTIONS );
-    const toolbarSwitch = new PreferencesToggleSwitch( toolbarEnabledProperty, false, true, {
+    const toolbarEnabledSwitch = new PreferencesToggleSwitch( toolbarEnabledProperty, false, true, {
       labelNode: quickAccessLabel,
-      a11yLabel: toolbarLabelString
+      a11yLabel: toolbarLabelString,
+      tandem: options.tandem.createTandem( 'toolbarEnabledSwitch' )
     } );
 
     // Speech output levels
@@ -129,9 +136,15 @@
       align: 'left',
       spacing: 5,
       children: [
-        createCheckbox( objectDetailsLabelString, audioModel.voicingObjectResponsesEnabledProperty ),
-        createCheckbox( contextChangesLabelString, audioModel.voicingContextResponsesEnabledProperty ),
-        createCheckbox( helpfulHintsLabelString, audioModel.voicingHintResponsesEnabledProperty )
+        createCheckbox( objectDetailsLabelString, audioModel.voicingObjectResponsesEnabledProperty,
+          options.tandem.createTandem( 'voicingObjectResponsesEnabledCheckbox' )
+        ),
+        createCheckbox( contextChangesLabelString, audioModel.voicingContextResponsesEnabledProperty,
+          options.tandem.createTandem( 'voicingContextResponsesEnabledCheckbox' )
+        ),
+        createCheckbox( helpfulHintsLabelString, audioModel.voicingHintResponsesEnabledProperty,
+          options.tandem.createTandem( 'voicingHintResponsesEnabledCheckbox' )
+        )
       ]
     } );
 
@@ -167,7 +180,7 @@
       voicingNameResponse: customizeVoiceString,
 
       // phet-io
-      tandem: Tandem.OPT_OUT
+      tandem: options.tandem.createTandem( 'expandCollapseButton' )
     } );
 
     const voiceOptionsContainer = new Node( {
@@ -186,13 +199,13 @@
     voiceOptionsLabel.addInputListener( voiceOptionsPressListener );
 
     const content = new Node( {
-      children: [ speechOutputContent, toolbarSwitch, voiceOptionsContainer, voiceOptionsContent ]
+      children: [ speechOutputContent, toolbarEnabledSwitch, voiceOptionsContainer, voiceOptionsContent ]
     } );
 
     // layout for section content, custom rather than using a LayoutBox because the voice options label needs
     // to be left aligned with other labels, while the ExpandCollapseButton extends to the left
-    toolbarSwitch.leftTop = speechOutputContent.leftBottom.plusXY( 0, 20 );
-    voiceOptionsLabel.leftTop = toolbarSwitch.leftBottom.plusXY( 0, 20 );
+    toolbarEnabledSwitch.leftTop = speechOutputContent.leftBottom.plusXY( 0, 20 );
+    voiceOptionsLabel.leftTop = toolbarEnabledSwitch.leftBottom.plusXY( 0, 20 );
     expandCollapseButton.leftCenter = voiceOptionsLabel.rightCenter.plusXY( 10, 0 );
     voiceOptionsContent.leftTop = voiceOptionsLabel.leftBottom.plusXY( 0, 10 );
     voiceOptionsOpenProperty.link( open => { voiceOptionsContent.visible = open; } );
@@ -201,7 +214,7 @@
     expandCollapseButton.focusHighlight = new FocusHighlightFromNode( voiceOptionsContainer );
 
     super( {
-      titleNode: voicingSwitch,
+      titleNode: voicingEnabledSwitch,
       contentNode: content
     } );
 
@@ -251,7 +264,10 @@
         voiceList = englishVoices.slice( 0, 12 );
       }
 
-      voiceComboBox = new VoiceComboBox( voiceList, audioModel.voiceProperty, phet.joist.sim.topLayer );
+      // phet-io - for when creating the Archetype for the Capsule housing the preferencesDialog, we don't have a sim global.
+      const parent = phet.joist.sim.topLayer || new Node();
+
+      voiceComboBox = new VoiceComboBox( voiceList, audioModel.voiceProperty, parent );
       voiceOptionsContent.addChild( voiceComboBox );
     };
     voicingManager.voicesChangedEmitter.addListener( voicesChangedListener );
@@ -297,7 +313,7 @@
  * @param {BooleanProperty} property
  * @returns {Checkbox}
  */
-const createCheckbox = ( labelString, property ) => {
+const createCheckbox = ( labelString, property, tandem ) => {
   const labelNode = new Text( labelString, PreferencesDialog.PANEL_SECTION_CONTENT_OPTIONS );
   return new Checkbox( labelNode, property, {
 
@@ -309,7 +325,7 @@
     voicingNameResponse: labelString,
 
     // phet-io
-    tandem: Tandem.OPT_OUT
+    tandem: tandem
   } );
 };
 
@@ -399,8 +415,18 @@
    * @param {SpeechSynthesisVoice[]} voices - list of voices to include from the voicingManager
    * @param {Property.<SpeechSynthesisVoice|null>} voiceProperty
    * @param {Node} parentNode - node that acts as a parent for the ComboBox list
+   * @param {Object} [options]
    */
-  constructor( voices, voiceProperty, parentNode ) {
+  constructor( voices, voiceProperty, parentNode, options ) {
+
+    options = merge( {
+      listPosition: 'above',
+      accessibleName: voiceLabelString,
+
+      // phet-io, opt out because we would need to instrument voices, but those could change between runtimes.
+      tandem: Tandem.OPT_OUT
+    }, options );
+
     const items = [];
 
     if ( voices.length === 0 ) {
@@ -421,13 +447,7 @@
     // voices
     voiceProperty.set( items[ 0 ].value );
 
-    super( items, voiceProperty, parentNode, {
-      listPosition: 'above',
-      accessibleName: voiceLabelString,
-
-      // phet-io
-      tandem: Tandem.OPT_OUT
-    } );
+    super( items, voiceProperty, parentNode, options );
 
     // voicing -  responses for the button should always come through, regardless of user selection of
     // responses. As of 10/29/21, ComboBox will only read the name response (which are always read regardless)
Index: joist/js/preferences/PreferencesDialog.js
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/joist/js/preferences/PreferencesDialog.js b/joist/js/preferences/PreferencesDialog.js
--- a/joist/js/preferences/PreferencesDialog.js	(revision 6f617114eef353e72bea7319355af3d753269558)
+++ b/joist/js/preferences/PreferencesDialog.js	(date 1647382028110)
@@ -10,15 +10,14 @@
  */
 
 import EnumerationProperty from '../../../axon/js/EnumerationProperty.js';
-import EnumerationValue from '../../../phet-core/js/EnumerationValue.js';
 import Enumeration from '../../../phet-core/js/Enumeration.js';
+import EnumerationValue from '../../../phet-core/js/EnumerationValue.js';
 import merge from '../../../phet-core/js/merge.js';
 import PhetFont from '../../../scenery-phet/js/PhetFont.js';
-import { KeyboardUtils } from '../../../scenery/js/imports.js';
-import { Node } from '../../../scenery/js/imports.js';
-import { Text } from '../../../scenery/js/imports.js';
+import { KeyboardUtils, Node, Text } from '../../../scenery/js/imports.js';
 import Dialog from '../../../sun/js/Dialog.js';
 import HSeparator from '../../../sun/js/HSeparator.js';
+import Tandem from '../../../tandem/js/Tandem.js';
 import audioManager from '../audioManager.js';
 import joist from '../joist.js';
 import joistStrings from '../joistStrings.js';
@@ -83,6 +82,7 @@
 
       // phet-io
       phetioDynamicElement: true,
+      tandem: Tandem.REQUIRED,
 
       // pdom
       positionInPDOM: true
@@ -100,10 +100,14 @@
     const selectedTabProperty = new EnumerationProperty( PreferencesTab.GENERAL );
 
     // the set of tabs you can can click to activate a tab panel
-    const preferencesTabs = new PreferencesTabs( supportedTabs, selectedTabProperty );
+    const preferencesTabs = new PreferencesTabs( supportedTabs, selectedTabProperty, {
+      tandem: options.tandem.createTandem( 'preferencesTabs' )
+    } );
 
     // the panels of content with UI components to select preferences, only one is displayed at a time
-    const preferencesPanels = new PreferencesPanels( preferencesModel, supportedTabs, selectedTabProperty );
+    const preferencesPanels = new PreferencesPanels( preferencesModel, supportedTabs, selectedTabProperty, {
+      tandem: options.tandem.createTandem( 'preferencesPanels' )
+    } );
 
     // visual separator between tabs and panels - as long as the widest separated content, which may change with i18n
     const tabPanelSeparator = new HSeparator( Math.max( preferencesPanels.width, preferencesTabs.width ), { lineWidth: 1 } );
Index: joist/js/preferences/VisualPreferencesPanel.js
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/joist/js/preferences/VisualPreferencesPanel.js b/joist/js/preferences/VisualPreferencesPanel.js
--- a/joist/js/preferences/VisualPreferencesPanel.js	(revision 6f617114eef353e72bea7319355af3d753269558)
+++ b/joist/js/preferences/VisualPreferencesPanel.js	(date 1647381649236)
@@ -9,10 +9,8 @@
 
 import merge from '../../../phet-core/js/merge.js';
 import StringUtils from '../../../phetcommon/js/util/StringUtils.js';
-import { VoicingText } from '../../../scenery/js/imports.js';
-import { voicingUtteranceQueue } from '../../../scenery/js/imports.js';
-import { Node } from '../../../scenery/js/imports.js';
-import { Text } from '../../../scenery/js/imports.js';
+import { Node, Text, VoicingText, voicingUtteranceQueue } from '../../../scenery/js/imports.js';
+import Tandem from '../../../tandem/js/Tandem.js';
 import joist from '../joist.js';
 import joistStrings from '../joistStrings.js';
 import PreferencesDialog from './PreferencesDialog.js';
@@ -30,18 +28,25 @@
 
   /**
    * @param {Object} visualModel - see PreferencesManager
+   * @param {Object} [options]
    */
-  constructor( visualModel ) {
-    super( {
+  constructor( visualModel, options ) {
+
+    options = merge( {
 
       // pdom
       tagName: 'div',
       labelTagName: 'h2',
-      labelContent: 'Visual'
-    } );
+      labelContent: 'Visual',
+
+      // phet-io
+      tandem: Tandem.REQUIRED
+    }, options );
+
+    super( options );
 
     const label = new Text( interactiveHighlightsString, PreferencesDialog.PANEL_SECTION_LABEL_OPTIONS );
-    const toggleSwitch = new PreferencesToggleSwitch( visualModel.interactiveHighlightsEnabledProperty, false, true, {
+    const interactiveHighlightsEnabledSwitch = new PreferencesToggleSwitch( visualModel.interactiveHighlightsEnabledProperty, false, true, {
       labelNode: label,
       descriptionNode: new VoicingText( interactiveHighlightsDescriptionString, merge( {}, PreferencesDialog.PANEL_SECTION_CONTENT_OPTIONS, {
         readingBlockContent: StringUtils.fillIn( labelledDescriptionPatternString, {
@@ -49,11 +54,12 @@
           description: interactiveHighlightsDescriptionString
         } )
       } ) ),
-      a11yLabel: interactiveHighlightsString
+      a11yLabel: interactiveHighlightsString,
+      tandem: options.tandem.createTandem( 'interactiveHighlightsEnabledSwitch' )
     } );
 
     const panelSection = new PreferencesPanelSection( {
-      titleNode: toggleSwitch
+      titleNode: interactiveHighlightsEnabledSwitch
     } );
     this.addChild( panelSection );
 
Index: studio/js/studio.js
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/studio/js/studio.js b/studio/js/studio.js
--- a/studio/js/studio.js	(revision dbf2959c7687b35447832c60e981d6cfceb735f3)
+++ b/studio/js/studio.js	(date 1647383155552)
@@ -229,6 +229,11 @@
    */
   onPhetioElementAdded( phetioID, phetioElementMetadata, initialValue ) {
     assert && assert( this.initialized, 'Must initialize before usage' );
+
+    if ( phetioID === 'friction.general.view.navigationBar.preferencesButton.preferencesDialog.preferencesTabs' ) {
+      debugger;
+    }
+
     assert && assert( typeof phetioElementMetadata.phetioTypeName === 'string', 'phetioTypeName must be string' );
 
     if ( phetioElementMetadata.phetioIsArchetype ) {
Index: joist/js/preferences/PreferencesPanels.js
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/joist/js/preferences/PreferencesPanels.js b/joist/js/preferences/PreferencesPanels.js
--- a/joist/js/preferences/PreferencesPanels.js	(revision 6f617114eef353e72bea7319355af3d753269558)
+++ b/joist/js/preferences/PreferencesPanels.js	(date 1647382221563)
@@ -9,8 +9,9 @@
  * @author Jesse Greenberg (PhET Interactive Simulations)
  */
 
-import { AlignGroup } from '../../../scenery/js/imports.js';
-import { Node } from '../../../scenery/js/imports.js';
+import merge from '../../../phet-core/js/merge.js';
+import { AlignGroup, Node } from '../../../scenery/js/imports.js';
+import Tandem from '../../../tandem/js/Tandem.js';
 import joist from '../joist.js';
 import AudioPreferencesPanel from './AudioPreferencesPanel.js';
 import GeneralPreferencesPanel from './GeneralPreferencesPanel.js';
@@ -24,46 +25,59 @@
    * @param {PreferencesManager} preferencesModel
    * @param {PreferencesTab[]} supportedTabs - list of Tabs supported by this Dialog
    * @param {EnumerationDeprecatedProperty.<PreferencesTab>} selectedTabProperty
+   * @param {Object} [options]
    */
-  constructor( preferencesModel, supportedTabs, selectedTabProperty ) {
-    super();
+  constructor( preferencesModel, supportedTabs, selectedTabProperty, options ) {
+
+    options = merge( {
+      tandem: Tandem.REQUIRED
+    }, options );
+    super( options );
 
     const panelAlignGroup = new AlignGroup( {
       matchVertical: false
     } );
 
-    // @private {PreferencesPanel[]}
+    // @private {PreferencesPanelContainer[]}
     this.content = [];
 
     let generalPreferencesPanel = null;
     if ( supportedTabs.includes( PreferencesDialog.PreferencesTab.GENERAL ) ) {
-      generalPreferencesPanel = new GeneralPreferencesPanel( preferencesModel.generalModel );
+      generalPreferencesPanel = new GeneralPreferencesPanel( preferencesModel.generalModel, {
+        tandem: options.tandem.createTandem( 'generalPreferencesPanel' )
+      } );
       const generalBox = panelAlignGroup.createBox( generalPreferencesPanel );
       this.addChild( generalBox );
-      this.content.push( new PreferencesPanel( generalPreferencesPanel, PreferencesDialog.PreferencesTab.GENERAL ) );
+      this.content.push( new PreferencesPanelContainer( generalPreferencesPanel, PreferencesDialog.PreferencesTab.GENERAL ) );
     }
 
     let visualPreferencesPanel = null;
     if ( supportedTabs.includes( PreferencesDialog.PreferencesTab.VISUAL ) ) {
-      visualPreferencesPanel = new VisualPreferencesPanel( preferencesModel.visualModel );
+      visualPreferencesPanel = new VisualPreferencesPanel( preferencesModel.visualModel, {
+        tandem: options.tandem.createTandem( 'visualPreferencesPanel' )
+      } );
       const visualBox = panelAlignGroup.createBox( visualPreferencesPanel );
       this.addChild( visualBox );
-      this.content.push( new PreferencesPanel( visualPreferencesPanel, PreferencesDialog.PreferencesTab.VISUAL ) );
+      this.content.push( new PreferencesPanelContainer( visualPreferencesPanel, PreferencesDialog.PreferencesTab.VISUAL ) );
     }
 
     let audioPreferencesPanel = null;
     if ( supportedTabs.includes( PreferencesDialog.PreferencesTab.AUDIO ) ) {
-      audioPreferencesPanel = new AudioPreferencesPanel( preferencesModel.audioModel, preferencesModel.toolbarEnabledProperty );
+      audioPreferencesPanel = new AudioPreferencesPanel( preferencesModel.audioModel, preferencesModel.toolbarEnabledProperty, {
+        tandem: options.tandem.createTandem( 'audioPreferencesPanel' )
+      } );
       const audioBox = panelAlignGroup.createBox( audioPreferencesPanel );
       this.addChild( audioBox );
-      this.content.push( new PreferencesPanel( audioPreferencesPanel, PreferencesDialog.PreferencesTab.AUDIO ) );
+      this.content.push( new PreferencesPanelContainer( audioPreferencesPanel, PreferencesDialog.PreferencesTab.AUDIO ) );
     }
 
     let inputPreferencesPanel = null;
     if ( supportedTabs.includes( PreferencesDialog.PreferencesTab.INPUT ) ) {
-      inputPreferencesPanel = new InputPreferencesPanel( preferencesModel.inputModel );
+      inputPreferencesPanel = new InputPreferencesPanel( preferencesModel.inputModel, {
+        tandem: options.tandem.createTandem( 'inputPreferencesPanel' )
+      } );
       this.addChild( inputPreferencesPanel );
-      this.content.push( new PreferencesPanel( inputPreferencesPanel, PreferencesDialog.PreferencesTab.INPUT ) );
+      this.content.push( new PreferencesPanelContainer( inputPreferencesPanel, PreferencesDialog.PreferencesTab.INPUT ) );
     }
 
     this.selectedTabProperty = selectedTabProperty;
@@ -79,7 +93,7 @@
 
   /**
    * @private
-   * @returns {PreferencesPanel} - the currently selected preferences panel
+   * @returns {PreferencesPanelContainer} - the currently selected preferences panel
    */
   getSelectedContent() {
     for ( let i = 0; i < this.content.length; i++ ) {
@@ -95,7 +109,7 @@
   /**
    * Focus the selected panel. The panel should not be focusable until this is requested, so it is set to be
    * focusable before the focus() call. When focus is removed from the panel, it should become non-focusable
-   * again. That is handled in PreferencesPanel class.
+   * again. That is handled in PreferencesPanelContainer class.
    * @public
    */
   focusSelectedPanel() {
@@ -119,7 +133,7 @@
  * An inner class that manages the panelContent and its value. A listener as added to the panel so that
  * whenever focus is lost from the panel, it is removed from the traversal order.
  */
-class PreferencesPanel extends Node {
+class PreferencesPanelContainer extends Node {
 
   /**
    * @param {Node} panelContent
Index: joist/js/Sim.js
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/joist/js/Sim.js b/joist/js/Sim.js
--- a/joist/js/Sim.js	(revision 6f617114eef353e72bea7319355af3d753269558)
+++ b/joist/js/Sim.js	(date 1647381917199)
@@ -529,7 +529,9 @@
       assert && assert( !options.simDisplayOptions.preferencesManager );
       options.simDisplayOptions.preferencesManager = this.preferencesManager;
 
-      this.toolbar = new Toolbar( this );
+      this.toolbar = new Toolbar( this, {
+        tandem: Tandem.GENERAL_VIEW.createTandem( 'toolbar' )
+      } );
 
       // when the Toolbar positions update, resize the sim to fit in the available space
       this.toolbar.rightPositionProperty.lazyLink( () => {
@@ -847,19 +849,18 @@
               // If the sim is in playback mode, then flush the timer's listeners. This makes sure that anything kicked
               // to the next frame with `timer.runOnNextTick` during startup (like every notification about a PhET-iO
               // instrumented element in phetioEngine.phetioObjectAdded()) can clear out before beginning playback.
-              if ( phet.joist.playbackModeEnabledProperty.value ) {
-                let beforeCounts = null;
-                if ( assert ) {
-                  beforeCounts = Array.from( Random.allRandomInstances ).map( n => n.numberOfCalls );
-                }
+              // if ( phet.joist.playbackModeEnabledProperty.value ) {
+              let beforeCounts = null;
+              if ( assert ) {
+                beforeCounts = Array.from( Random.allRandomInstances ).map( n => n.numberOfCalls );
+              }
 
-                stepTimer.emit( 0 );
+              stepTimer.emit( 0 );
 
-                if ( assert ) {
-                  const afterCounts = Array.from( Random.allRandomInstances ).map( n => n.numberOfCalls );
-                  assert && assert( _.isEqual( beforeCounts, afterCounts ),
-                    `Random was called more times in the playback sim on startup, before: ${beforeCounts}, after: ${afterCounts}` );
-                }
+              if ( assert ) {
+                const afterCounts = Array.from( Random.allRandomInstances ).map( n => n.numberOfCalls );
+                assert && assert( _.isEqual( beforeCounts, afterCounts ),
+                  `Random was called more times in the playback sim on startup, before: ${beforeCounts}, after: ${afterCounts}` );
               }
 
               // After the application is ready to go, remove the splash screen and progress bar.  Note the splash
Index: joist/js/preferences/NavigationBarPreferencesButton.js
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/joist/js/preferences/NavigationBarPreferencesButton.js b/joist/js/preferences/NavigationBarPreferencesButton.js
--- a/joist/js/preferences/NavigationBarPreferencesButton.js	(revision 6f617114eef353e72bea7319355af3d753269558)
+++ b/joist/js/preferences/NavigationBarPreferencesButton.js	(date 1647383204788)
@@ -9,6 +9,8 @@
 import merge from '../../../phet-core/js/merge.js';
 import { Path } from '../../../scenery/js/imports.js';
 import userCogSolidShape from '../../../sherpa/js/fontawesome-5/userCogSolidShape.js';
+import Dialog from '../../../sun/js/Dialog.js';
+import PhetioCapsule from '../../../tandem/js/PhetioCapsule.js';
 import Tandem from '../../../tandem/js/Tandem.js';
 import joist from '../joist.js';
 import JoistButton from '../JoistButton.js';
@@ -32,15 +34,19 @@
       maxWidth: 25
     } );
 
-    let preferencesDialog = null;
-    super( icon, backgroundColorProperty, Tandem.OPT_OUT, {
+    assert && assert( !options.listener, 'PhetButton sets listener' );
+    const preferencesDialogCapsule = new PhetioCapsule( tandem => {
+      return new PreferencesDialog( preferencesModel, {
+        tandem: tandem
+      } );
+    }, [], {
+      tandem: options.tandem.createTandem( 'preferencesDialogCapsule' ),
+      phetioType: PhetioCapsule.PhetioCapsuleIO( Dialog.DialogIO )
+    } );
+
+    super( icon, backgroundColorProperty, options.tandem, {
       listener: () => {
-        if ( !preferencesDialog ) {
-          preferencesDialog = new PreferencesDialog( preferencesModel, {
-            tandem: Tandem.OPT_OUT
-          } );
-        }
-
+        const preferencesDialog = preferencesDialogCapsule.getElement();
         preferencesDialog.show();
         preferencesDialog.focusSelectedTab();
       },
Index: phet-io/js/phetioEngine.js
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/phet-io/js/phetioEngine.js b/phet-io/js/phetioEngine.js
--- a/phet-io/js/phetioEngine.js	(revision acc922179f54866e69ae2c7debfcd7de6029b1ee)
+++ b/phet-io/js/phetioEngine.js	(date 1647382659533)
@@ -34,8 +34,8 @@
 import dotRandom from '../../dot/js/dotRandom.js';
 import packageJSON from '../../joist/js/packageJSON.js';
 import SimInfo from '../../joist/js/SimInfo.js';
-import { Mouse } from '../../scenery/js/imports.js';
 import arrayRemove from '../../phet-core/js/arrayRemove.js';
+import { Mouse } from '../../scenery/js/imports.js';
 import EventType from '../../tandem/js/EventType.js';
 import phetioAPIValidation from '../../tandem/js/phetioAPIValidation.js';
 import PhetioObject from '../../tandem/js/PhetioObject.js';
@@ -664,6 +664,9 @@
 
     // Populate the map
     this.phetioObjectMap[ phetioID ] = phetioObject;
+    if ( phetioID === 'friction.general.view.navigationBar.preferencesButton.preferencesDialog.preferencesTabs' ) {
+      debugger;
+    }
 
     /**
      * After the PhetioObject construction is finished, notify listeners that it is ready for interoperability.

```

**Comment by zepumph** [comment]
Alright. The view is committed above. It still isn't totally clear how we will have the model instrumented, since we largely don't want preferences to persist. Even so, this seems like a good commit point. I will work to see what more we would want to do with the model.

**Comment by pixelzoom** [comment]
fourier-making-waves is a sim that is partially instrument.  It suddenly started failing CT with:

<details>
<summary>Stack Trace</summary>

fourier-making-waves : fuzz : unbuilt
https://bayes.colorado.edu/continuous-testing/ct-snapshots/1648595034925/fourier-making-waves/fourier-making-waves_en.html?continuousTest=%7B%22test%22%3A%5B%22fourier-making-waves%22%2C%22fuzz%22%2C%22unbuilt%22%5D%2C%22snapshotName%22%3A%22snapshot-1648595034925%22%2C%22timestamp%22%3A1648601523851%7D&brand=phet&ea&fuzz&memoryLimit=1000
Query: brand=phet&ea&fuzz&memoryLimit=1000
Uncaught Error: Assertion failed: AquaRadioButton tandem.name must end with RadioButton: fourierMakingWaves.discreteScreen.view.controlPanel.graphControlsSubpanel.seriesRadioButtonGroup.placeHolder725597
Error: Assertion failed: AquaRadioButton tandem.name must end with RadioButton: fourierMakingWaves.discreteScreen.view.controlPanel.graphControlsSubpanel.seriesRadioButtonGroup.placeHolder725597
at window.assertions.assertFunction (https://bayes.colorado.edu/continuous-testing/ct-snapshots/1648595034925/assert/js/assert.js:25:13)
at new AquaRadioButton (https://bayes.colorado.edu/continuous-testing/ct-snapshots/1648595034925/chipper/dist/js/sun/js/AquaRadioButton.js:55:15)
at new AquaRadioButtonGroup (https://bayes.colorado.edu/continuous-testing/ct-snapshots/1648595034925/chipper/dist/js/sun/js/AquaRadioButtonGroup.js:71:27)
at new HorizontalAquaRadioButtonGroup (https://bayes.colorado.edu/continuous-testing/ct-snapshots/1648595034925/chipper/dist/js/sun/js/HorizontalAquaRadioButtonGroup.js:18:5)
at new SeriesTypeRadioButtonGroup (https://bayes.colorado.edu/continuous-testing/ct-snapshots/1648595034925/chipper/dist/js/fourier-making-waves/js/common/view/SeriesTypeRadioButtonGroup.js:46:5)
at new GraphControlsSubpanel (https://bayes.colorado.edu/continuous-testing/ct-snapshots/1648595034925/chipper/dist/js/fourier-making-waves/js/discrete/view/DiscreteControlPanel.js:273:40)
at new DiscreteControlPanel (https://bayes.colorado.edu/continuous-testing/ct-snapshots/1648595034925/chipper/dist/js/fourier-making-waves/js/discrete/view/DiscreteControlPanel.js:71:51)
at new DiscreteScreenView (https://bayes.colorado.edu/continuous-testing/ct-snapshots/1648595034925/chipper/dist/js/fourier-making-waves/js/discrete/view/DiscreteScreenView.js:188:26)
at DiscreteScreen.createView (https://bayes.colorado.edu/continuous-testing/ct-snapshots/1648595034925/chipper/dist/js/fourier-making-waves/js/discrete/DiscreteScreen.js:36:18)
at DiscreteScreen.initializeView (https://bayes.colorado.edu/continuous-testing/ct-snapshots/1648595034925/chipper/dist/js/joist/js/Screen.js:210:23)
id: Bayes Chrome
Snapshot from 3/29/2022, 5:03:54 PM

</details>

SeriesTypeRadioButtonGroup is instrumented, but its items do not have tandemName options specified yet. That hasn’t been a problem up until now.

The culprit is in AquaRadioButtonGroup, added today at 4:41PM in https://github.com/phetsims/sun/commit/86507bb96af96b3540df90308deb47fcd8a1879a for this issue:

```
          // Instead of using Tandem.REQUIRED, use the same tandem that is passed into the group, helping to support Tandem.OPT_OUT
          tandem: options.tandem.createTandem( item.tandemName || 'placeHolder' + dotRandom.nextInt( 1000000 ) )
```

This code is violating the requirement that tandemName must end with "RadioButton" suffix.

**Comment by zepumph** [comment]
Makes sense. Thanks for catching it. Fixed above.

**Comment by pixelzoom** [comment]
It makes me sad that this "RadioButton" suffix convention is now hardcoded in at least 201 places -- search for "RadioButton'" and "RadioButton`".   Should it be factored out?   Or should radio-button groups take `tandemNamePrefix` instead of `tandemName`, and automatically add the "RadioButton" suffix?

**Comment by pixelzoom** [comment]
Moving the tandem name convention questions to https://github.com/phetsims/phet-io/issues/1855.

**Comment by arouinfar** [comment]
During quarterly planning, @zepumph indicated that the Preferences dialog is ready for design review. I reviewed mostly in Friction, with some cross-referencing in John Travoltage and Energy Skate Park (has a General tab, but nothing else).

> During PhET-iO meeting today, we decided that preferences Properties would not be stateful.

I’m not sure where this work is at, but this is only partially true for the Audio settings. The Audio Features toggle, Sounds toggle, and Extra Sounds checkbox are all stateful. This seems appropriate for the Audio Features and Sounds toggles because they are connected to the sound icon in the navbar, but doesn’t seem desirable for Extra Sounds.

> In general, we want to customize per feature, and not internal to a feature. So you can't hide a checkbox in the Voicing controls, but you can hide the whole voicing section.

This task is still unchecked, but I wanted to also add that the top-level visible properties for these features should be `phetioFeatured: true` and assuming the children all become `phetioReadOnly: true` we should also make them `phetioFeatured: false`. There are checkboxes, toggle switches, etc. that come with their `enabledProperty` and `visibleProperty`  `phetioFeatured: true` by default, and we should make sure to undo that if they aren’t actually editable.

> Ability to hide tabs in the preferences, can we use the QP below, or do we need new Properties that are instrumented.

The current tree structure splits the dialog into two sections: `preferencesPanels` and `preferencesTabs`, each containing a top-level visibleProperty that hides all tab names/contents (which seems weird) and sub-sections for General, Visual, Audio, etc. This results in a clunky flow when hiding a tab and its contents. The client needs to set both `preferencesTabs.*Tab.visibleProperty` AND the corresponding `preferencesPanels.*PreferencesPanel.visibleProperty` to false. 

Using query parameters to hide the tabs seems reasonable, though I don’t know if it would be applicable to all tabs, such as General. An alternative might be to group the dialog contents by type (with a top-level `visibleProperty`) and then divide into tab/panel. Here’s a rough mockup – the tandem names are stand-ins, we don’t necessarily need to repeat the category name in the children.
```
- preferencesDialog
     - generalPreferences
          visibleProperty
          + generalTab
          + generalPanel
     + audioPreferences
     + visualPreferences
    ...
``` 


Some additional comments:
*  `selectedTabProperty` includes options for all possible tabs, not just the ones available within a particular sim. Should irrelevant tabs be hidden from the selection options? Selecting an irrelevant tab (such as “Input”) doesn’t appear to crash the sim or throw an assertion, but it also doesn’t seem like a desirable customization. 
* Under `audioPreferencesPanel`, there are two subsections for Voicing and Sound both containing controls for their respective features – `voicingPanelSection` and `soundPanelSection`. These sections could be renamed something more streamlined like `voicingControls` and `soundControls`. This is a pattern we often use when a sim has a large control panel with subsections, for example `moleculePolarity.twoAtomsScreen.view.controlPanel`.

* Is it possible to make `isShowingProperty` for ALL dialogs `phetioFeatured: true`? Clients may want to know if/when students have a dialog open.

**Comment by zepumph** [comment]
>Is it possible to make isShowingProperty for ALL dialogs phetioFeatured: true? Clients may want to know if/when students have a dialog open.


Yes indeed! Updated above. It is still `phetioReadOnly:true`.

**Comment by zepumph** [comment]
I think that some of the next steps may be helped by a design meeting. But I will take a look at hiding individual preferences sections/tabs first.

**Comment by zepumph** [comment]
This is still blocking Friction

**Comment by zepumph** [comment]
From a design meeting with @arouinfar today:

* We discussed that the general sab of the PreferencesDialog is really its own beast, that should be a lot more customizable than other tabs. We mentioned that perhaps character sets and language preferences should be put into a separate and new tab instead of in general (since they aren't sim specific).
* We discussed query parameters generally, and their use and benefit in phet-io. In general they feel valuable if they should be public to phet brand too, but otherwise, they feel like more of a workaround in PhET-iO brand. 
* In general, PreferencesModel can have linkedElements, but they shouldbe not readonly but should be phetioState:false. This feels arbitrary to AR and I,but good enough for a first pass.
*  Uninstrument these properties: `friction.general.view.navigationBar.preferencesButton.preferencesDialogCapsule.preferencesDialog.preferencesPanels.visibleProperty` 
`friction.general.view.navigationBar.preferencesButton.preferencesDialogCapsule.preferencesDialog.preferencesTabs.visibleProperty`
* Uninstrument all view components that run on a Model property that we will make sure is instrumented.  In the general tab, those will have custom context and instrumentation practices.
* `audioManager.audioEnabledProperty` is allowed to be stateful
* `soundManager.enabledProperty` and `soundManager.extraSoundsEnabledProperty` should both be not stateful.
*

**Comment by zepumph** [comment]
>Uninstrument all view components that run on a Model property that we will make sure is instrumented. In the general tab, those will have custom context and instrumentation practices.

I am going to link to this comment in code comments to help track the decision of using Tandem.OPT_OUT. @arouinfar and I decided that while the underlying Properties may be readonly and available for PhET-iO, there is no reason to instrument the controlling view components. We wouldn't want to hide preferences (besides general) from the user. That is antethetical to the usability goals of PhET preferences in general, and we don't want PhET-iO clients to have that power.

**Comment by zepumph** [comment]
Ok. I removed the View instrumentation. I also removed the visibleProperty from the general panel. Please review the preferences dialog and the new general.model.preferencesModel.

**Comment by zepumph** [comment]
A couple more questions for @arouinfar :

- should the `selectedTabProperty` be unstateful and readOnly also? I'm pretty sure, but since it lives in the view I thought I'd ask
- I also opted out for the localization and character set items since they were in the preferences/ folder too. This way we have the same behavior for all settings but the sim-specific components on the general tab, but let me know if you feel differently.

**Comment by arouinfar** [comment]
@zepumph in general, the tree looks good, but I ran into one issue:

* The contents of `general.view.toolbar` should likely all be read-only. It seems odd for users to be in control of the `visibleProperty` of the toolbar and its contents.

> Please review the preferences dialog and the new general.model.preferencesModel.

This all looks good to me @zepumph!

> should the selectedTabProperty be unstateful and readOnly also? I'm pretty sure, but since it lives in the view I thought I'd ask

Yes, that sounds reasonable to me.

> I also opted out for the localization and character set items since they were in the preferences/ folder too

Eventually the Localization Settings will include a UI control for users to select the sim locale within the all file. We also tell clients to use the `locale` query parameter to set the sim language, so I think these controls might need to be stateful. (Or maybe just locale?)

**Comment by samreid** [comment]
Today @zepumph and I converted Gravity and Orbits to use PreferencesConfiguration.  @arouinfar can you please spot check it and make sure it seems good?

**Comment by samreid** [comment]
I addressed several of the recommendations above:

> Eventually the Localization Settings will include a UI control for users to select the sim locale within the all file. We also tell clients to use the locale query parameter to set the sim language, so I think these controls might need to be stateful.

When this is introduced, the `localeProperty` will be instrumented and stateful and the UI components will not need to be stateful.  

I think that's everything for this issue.  @arouinfar can you please review, and also check the new Preferences Dialog for Gravity and Orbits?  Please close if all is well.

**Comment by arouinfar** [comment]
@samreid the Projector Mode checkbox behaves as expected and looks fine in the Studio tree. However, the font is huge. It looks like it might match the size of "General" but it should instead be a more appropriate size for the contents. For example, refer to the "Extra Sounds" checkbox (found in Friction, but likely a common code thing).
<img width="256" alt="image" src="https://user-images.githubusercontent.com/8419308/183778812-1800a72a-bce1-428c-b4b2-42b4914db91b.png">

<img width="267" alt="image" src="https://user-images.githubusercontent.com/8419308/183778753-74f5e42e-0aa6-4438-9e3d-ecafc75873ad.png">

Just an FYI, the inclusive features blurb at the bottom of the General tab may soon become optional, see https://github.com/phetsims/joist/issues/814. I would consider this blocking for any PhET-iO sims that have a Preferences dialog, but no inclusive features (like GAO).

**Comment by samreid** [comment]
I reduced the size and added a commit in https://github.com/phetsims/joist/issues/814 that omits the inclusive features blurb at the bottom of the General tab if there is only one tab.

Ready for review.

**Comment by arouinfar** [comment]
Thanks @samreid, the visual changes you made in GAO seem fine, but I agree with your comments in https://github.com/phetsims/joist/issues/814#issuecomment-1210161680 which we can discuss in design meeting tomorrow.

It looks like the Projector Mode checkbox is no longer instrumented, and it seems like the entire Preferences Dialog may have been uninstrumented. I can still find the capsule in the tree, but it doesn't have any children even after opening the dialog. (All Elements selected.) 
<img width="239" alt="image" src="https://user-images.githubusercontent.com/8419308/183942978-114bb406-f317-4cea-9d48-4d1e9434a5dc.png">

I also noticed that we haven't featured `preferencesButton.visibleProperty` which is inconsistent with the audio and keyboard help buttons.

These issues aren't sim-specific so back to @samreid and @zepumph.

**Comment by samreid** [comment]
I featured the preferencesButton.visibleProperty and fixed a bug that prevented the subtree from in elements from showing in https://github.com/phetsims/studio/issues/268.  @arouinfar can you please re-review?

**Comment by samreid** [comment]
In today's meeting, 

### Regarding saving state with or without preferences
@JacquiHayes identified that privacy laws may forbid detecting what a11y preferences are selected without specialized permissions.  
@kathy-phet points out that the PhET-iO clients may want to have this flexibility, like being able to save a state without these values, or another API call that also gives these values.
@zepumph asks if we could be liable by enabling this API?  The team thinks the user of that API would be liable.
@JacquiHayes has not heard from clients about this.
@zepumph indicates that we can do this release without these features, but planning to add an option for this in the future.  Adding "2 kinds of state" would lead to increased complexity in the future.  Especially if we don't have a client that needs that at all.
@kathy-phet: But users would have to re-enable the preferences they want, but maybe that is OK.  But what if a student is working on a sim in a wrapper. The wrapper saves progress.  When the student comes back the next day, it would have forgotten their usability preferences.
@zepumph added a dev tool for localStorage to save the preferences.  Would it work for your case?
@samreid: I don't think so
@zepumph also, we didn't want preferences to persist for cases like computer labs, etc.
@kathy-phet says maybe there should be a separate API call just to get the preferences.

So we can have `getState` vs `getStateWithPreferences` OR we can have `getState` + `getPreferencesState`.  It is unclear which is preferable.

**Comment by zepumph** [comment]
Still on me, pinged during today's PhET-iO meeting.

**Comment by zepumph** [comment]
I'd like to co-assign @samreid because my time pre-vacation is running short, and I don't want to be a bottle neck. I think that @samreid could definitely work on marking things as phetioReadOnly: true again, and likely also the simSoundEnabledProperty rename. I'm still trying to get to this though!

**Comment by samreid** [comment]
@marlitas and I implemented the first step by making things no longer read only. We have no confidence that we have covered model elements that do not appear in gravity and orbits, and we are considering adding a post start up a session check that all nested elements under the preferences model are marked as read only false.  We think this would take 30 minutes or so but would protect occurrences outside gravity and orbits, and all future occurrences.  But if one day, we decide to make even one preferences model property as `phetioReadOnly: true` then we would have to discard the assertion, so it may have limited use.

**Comment by samreid** [comment]
I completed the assertions that @marlitas and I started.  I moved the remaining issue to https://github.com/phetsims/joist/issues/864.  I think this issue can be closed.  We may discover details during design review or QA testing, in that case, we can reopen or create side issues.

**Comment by zepumph** [comment]
Looks like `voiceVolumeProperty` should not be instrumented.

**Comment by zepumph** [comment]
Just a couple more tweaks in voicing implementation from a sloppy first pass (sorry!)
